packages/catalog-client/src/CatalogClient.ts

```diff
@@ -490,9 +491,10 @@ export class CatalogClient implements CatalogApi {
     options?: CatalogRequestOptions,
   ): Promise<T> {
     const url = `${await this.discoveryApi.getBaseUrl('catalog')}${path}`;
-    const headers: Record<string, string> = options?.token
-      ? { Authorization: `Bearer ${options.token}` }
-      : {};
+    const headers: Record<string, string> = { ...options?.headers };
+    if (options?.token) {
+      headers.Authorization = `Bearer ${options.token}`;
+    }
```

Comment:
maybe we could "flip" the logic to make it possible to overcustomize the Authorization header:

```suggestion
    const headers: Record<string, string> = {
      ...(!!options?.token && { Authorization: `Bearer ${options.token}` }),
      ...options?.headers,
    };
```

Reply:
My thought was that if you specify an authorization header under headers and a token, the token would take precedence since that's the current behaviour

Reply:
and you are perfectly right! Ignore